### ISSUE TITLE
feature_learning_testing

### ISSUE THREAD

**Issue by ymcidence** [richzhang/colorization#19]
Hi, Richard,

For the feature-learning-testing segmentation part, you firstly pre-train an AlexNet networkon the colourization task, and then fine-tune a FCN-AlexNet using the pre-trianed AlexNet model for segmentation on PASCAL VOC 2012, is it right?

If yes, however, I find the pre-trained model in the file solve.py for segmentation
 "default='./models/alexnet_release_450000_nobn_rs.caffemodel')".
I am not sure the model "alexnet_release_450000_nobn_rs.caffemodel" is trained for colourization because the file 'train_val_nobn_rs.prototxt' which looks related to the model from their names is a classification task. Anything I understand wrongly?

Thank you very much.

**Comment by richzhang** [comment]
Yes, so `train_val_nobn_rs.prototxt` is used for the rescaling portion of `postprocess_model.sh`, in particular in the rescaling command `python ./resources/magic_init/magic_init_mod.py ./train_alexnet/train_val_nobn_rs.prototxt ./train_alexnet/colornet_iter_${2}_nobn_rs.caffemodel -l ./train_alexnet/colornet_iter_${2}_nobn.caffemodel -nit 10 -cs --gpu ${1}`

The rescaling adjusts the weights so that all layers "learn" at the same rate roughly. It doesn't use any labels - it injects a random gradient at the top fc7 layer. You can see that the label blob in `train_val_nobn_rs.prototxt` is all 0's.
